service/s3.cc

```diff
@@ -303,6 +303,14 @@ struct S3Downloader {
             maxRqs = 5;
         chunks.resize(maxRqs);
 
+        /* Hack to ensure that the file's last modified time is earlier than 1
+           seconds before the start time of the download. Useful for the etag
+           check below. Should seldom be executed in practice. */
+        Date now = Date::now();
+        if (info.lastModified > now.plusSeconds(-1)) {
+            ML::sleep(1);
+        }
+
```

Comment:
The hack above enables to make sure that the file was created at a different second than the download start time, in order to avoid false negatives for the "lastModified" comparison below.